src/rgw/rgw_rados.cc

```diff
@@ -345,6 +345,24 @@ class RGWDataNotifierManager : public RGWCoroutinesManager {
     }
     return run(stacks);
   }
+
+  int notify2_all(map<rgw_zone_id, RGWRESTConn *>& conn_map,
+		 bc::flat_map<int, bc::flat_set<rgw_data_notify_entry> >& shards) {
```

Comment:
there should just be the one `notify_all()` function that spawns a single coroutine for each entry in `conn_map`

you'll need to define a new coroutine that first tries the notify2 API, and retries with v1 if v2 fails with `ERR_METHOD_NOT_ALLOWED`